docs: update certificate spec (#1240)
notaryproject/notation: specs/cmd/certificate.md

Error: required flag(s) "store", "type" not set
```

### Generate a local RSA key and a corresponding self-generated certificate for testing purpose

```bash
notation certificate generate-test "wabbit-networks.io"
```

> Upon successful execution, a local key file named `wabbit-networks.io.key` and a certificate file named `wabbit-networks.io.crt` are generated and stored in `$XDG_CONFIG_HOME/notation/localkeys/`. `wabbit-networks.io` is also used as the certificate's subject.CommonName. The certificate is added to trust store `wabbit-networks.io` of type `ca`. And the key with name `wabbit-networks.io` is added into `{NOTATION_CONFIG}/signingkeys.json`.

Reviewer: ```suggestion
Upon successful execution, a local key file named `wabbit-networks.io.key` and a certificate file named `wabbit-networks.io.crt` are generated and stored in `$XDG_CONFIG_HOME/notation/localkeys/`. `wabbit-networks.io` is also used as the certificate's subject.CommonName. The certificate is added to trust store `wabbit-networks.io` of type `ca`. An entry containing the file paths for the key and certificate is added to `{NOTATION_CONFIG}/signingkeys.json`.
```

Author: Updated with a revise.

### Clean up a test RSA key and its corresponding certificate that were generated using the "generate-test" command

> Use the following command to clean up a test RSA key and its corresponding certificate that were generated using the `generate-test` command.

Reviewer: The following is an example, as it uses an example common name.

Author: Removed this line as it's repeating line 259.

Error: required flag(s) "store", "type" not set
```

### Generate a local RSA key and a corresponding self-generated certificate for testing purpose

```bash
notation certificate generate-test "wabbit-networks.io"
```

Upon successful execution, a local key file named `wabbit-networks.io.key` and a certificate file named `wabbit-networks.io.crt` are generated and stored in `$XDG_CONFIG_HOME/notation/localkeys/`. `wabbit-networks.io` is also used as the certificate's subject.CommonName. The certificate is added to trust store `wabbit-networks.io` of type `ca`. An entry with name `wabbit-networks.io` containing the file paths of the key and certificate is added to `{NOTATION_CONFIG}/signingkeys.json`.

### Clean up a test RSA key and its corresponding certificate that were generated using the "generate-test" command

```bash
notation certificate cleanup-test "wabbit-networks.io"
```

A prompt will be displayed, asking the user to confirm the cleanup.

```text
Are you sure you want to clean up test key <name> and its certificate? [y/N]

> ```

Reviewer 2: Since we are deleting stuffs that not recoverable, we should let the users know what will be deleted. If you think it is not consistent with `generate-test`, we can also add them in that command.

Author: In that case, it would be more helpful to put it into the help page, so that the user gets to know what they are doing before running the command.

Author: Updated.

A prompt will be displayed, asking the user to confirm the cleanup.

```text

> Are you sure you want to clean up test key <name> and its certificate? [y/N]

Reviewer: This is required by Shiwei, should we remove it?

Author: Yes, because this information is not provided to the user during `generate-test` process either.
Both commands print out this information at the final outputs.

A sample output for a successful execution:

```text

> Successfully deleted certificate <name>.crt from trust store <name> of type ca.

Reviewer: The original one is suggested by Copilot :)

Author: Apparently, Copilot does not have the big picture of Notation.
